src/import/importConfiguration.ts

```diff
@@ -536,8 +536,24 @@ function parseDependencies(
   if (yamlDependencies == null) {
     return;
   }
-  return Object.entries(yamlDependencies).map(([packageId, version]) => {
-    return { packageId, version: `${version}` };
+  return Object.entries(yamlDependencies).map(([packageId, versionOrDetails]) => {
+    if (typeof versionOrDetails === 'string' || typeof versionOrDetails === 'number') {
+      return { packageId, version: `${versionOrDetails}` };
+    } else if (versionOrDetails == null) {
+      // this is an error condition, but we'll just make it an incomplete dependency and
+      // handle the error later
+      return { packageId };
+    }
+    // else it's the complex object
+    return removeUndefinedValues({
+      id: versionOrDetails.id,
+      packageId,
+      uri: versionOrDetails.uri,
+      version:
+        typeof versionOrDetails.version === 'number'
+          ? `${versionOrDetails.version}`
+          : versionOrDetails.version
```

Comment:
I think to me the first way seems better. But is there real reason to not just put `undefined` instead of `versionOrDetails.version` in the else of the conditional, since we know `versionOrDetails.version` should be `undefined`, and if it isn't `undefined` then something crazy is happening.

Reply:
OK.  Sold!